src/osd/PGLog.h

```diff
@@ -1475,6 +1475,9 @@ struct PGLog : DoutPrefixProvider {
 	    ceph_assert(dups.back().version < dup.version);
 	  }
 	  dups.push_back(dup);
+	  if (dups.size() >= g_ceph_context->_conf->osd_pg_log_dups_tracked) {
+	    dups.pop_front();
```

Comment:
It is a bit unfortunate that on every start we will need to waste time (it was up to 10 minutes in our case) reading all dups unless the user runs ceph-objectstore-tool to trim the log manually. And the user will not have any clue what is going on.

May be we should add a counter here and log at least a warning message with the recommendation to trim the log manually, when the counter is large enough?

Reply:
@trociny if the osd sopped again, we will save the new duplicate log, so we will load the trimmed list next time.

Reply:
So all the old dup keys will be removed from rocksdb on the save? I was not aware of it. Then I think there is no problem.